packages/nocodb/src/lib/db/sql-client/lib/sqlite/SqliteClient.ts

```diff
@@ -1956,19 +1945,31 @@ class SqliteClient extends KnexClient {
     const defaultValue = getDefaultValue(n);
     let shouldSanitize = true;
     if (change === 2) {
-      query += this.genQuery(
-        `ALTER TABLE ?? RENAME COLUMN ?? TO ??`,
-        [t, o.cn, n.cn],
+      let q1 = this.genQuery(
+        `ALTER TABLE ?? RENAME COLUMN ?? TO ??;`,
+        [t, o.cn, `${o.cno}_old`],
         shouldSanitize
       );
+      
+      let q2 = ''
+      q2 += this.genQuery(` ADD ?? ${n.dt}`, [n.cn], shouldSanitize);
+      q2 += n.dtxp && n.dt !== 'text' ? `(${n.dtxp})` : '';
+      q2 += n.cdf ? (n.cdf.includes(',') ? ` DEFAULT ('${n.cdf}')` : ` DEFAULT ${n.cdf}`) : ' ';
+      q2 += n.rqd ? ` NOT NULL` : ' ';
+      q2 = this.genQuery(`ALTER TABLE ?? ${q2};`, [t], shouldSanitize);
+
+      let q3 = this.genQuery(`UPDATE ?? SET ?? = ??;`, [t, n.cn, `${o.cno}_old`], shouldSanitize);
+
+      let q4 = this.genQuery(`ALTER TABLE ?? DROP COLUMN ??;`, [t, `${o.cno}_old`], shouldSanitize);
```

Comment:
What does this `_old` hold ? was this part of our meta earlier. If not, we need to see its documented and uniform across other clients. Can discuss offline.

Reply:
Because of the limitations with alter on SQLite on first query we rename original column to `colName`_old then we recreate column and update all data using this backup column on next query (q4) we drop it.

Also I realized there should be a pragma foreign keys on | off surrounding the queries to support foreign keys, I will handle this situation as well.